Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -1001,15 +1001,75 @@ void Element::set_scroll_top(double y)
     block_container->set_scroll_offset(scroll_offset);
 }
 
+// https://w3c.github.io/csswg-drafts/cssom-view/#dom-element-scrollwidth
 int Element::scroll_width() const
 {
-    dbgln("FIXME: Implement Element::scroll_width() (called on element: {})", debug_description());
+    // 1. Let document be the element's node document.
+    auto const& document = this->document();
+
+    // 2. If document is not the active document, return zero and terminate these steps.
+    if (!document.is_active())
+        return 0;
+
+    // 3. Let viewport width be the width of the viewport excluding the width of the scroll bar,
+    //    if any, or zero if there is no viewport.
+    // FIXME: Check if there is a viewport
+    auto const viewport_width = document.browsing_context()->viewport_rect().width();
+
+    // 4. If the element is the root element and document is not in quirks mode
+    //    return max(viewport scrolling area width, viewport width).
+    if (is<HTML::HTMLHtmlElement>(*this) && !document.in_quirks_mode())
+        return max(document.browsing_context()->viewport_scroll_offset().x(), viewport_width);
+
+    // 5. If the element is the body element, document is in quirks mode
+    //    and the element is not potentially scrollable,
+    //    return max(viewport scrolling area width, viewport width).
+    if (is<HTML::HTMLBodyElement>(*this) && document.in_quirks_mode() && !this->is_potentially_scrollable())
+        return max(document.browsing_context()->viewport_scroll_offset().x(), viewport_width);
+
+    // 6. If the element does not have any associated box return zero and terminate these steps.
+    if (!this->layout_node() || !is<Layout::BlockContainer>(this->layout_node()))
+        return 0;
+
+    // FIXME: 7. Return the width of the element's scrolling area.
+    //       This may have to go in a separate PR, as it involves
+    //       CSS overflow implementation
```

Comment:
This meta-commentary about your PR and doesn't need to be in the code :)

Reply:
Oops,  thanks for the heads up. I changed it to a note that just warns about the need for a CSS overflow implementation before that logic can be implemented.